src/smart-components/workspaces/WorkspaceListTable.tsx

```diff
@@ -143,14 +205,18 @@ const WorkspaceListTable = () => {
                   onClick: () => {
                     navigate(paths['edit-workspaces-list'].link.replace(':workspaceId', workspace.id));
                   },
+                  isDisabled: !canModify(workspace, 'edit'),
+                },
+                {
+                  title: <Divider component="li" key="divider" />,
+                  isSeparator: true,
                 },
-                { title: <Divider component="li" key="divider" /> },
                 {
                   title: 'Delete workspace',
                   onClick: () => {
                     handleModalToggle([workspace]);
                   },
-                  isDisabled: workspace.children && workspace.children.length > 0,
+                  isDisabled: (workspace.children && workspace.children.length > 0) || !canModify(workspace, 'delete'),
                   isDanger: !(workspace.children && workspace.children.length > 0),
```

Comment:
We should also change this slightly so the disabled button is not of `danger` type.
```suggestion
                  isDanger: !(workspace.children && workspace.children.length > 0) && canModify(workspace, 'delete'),
```